File: docs/test-beds/csharp/fixture.cs
// C# test bed
using System;
using System.Collections.Generic;

namespace TestBed {
    public class Box<T> { public T Value { get; set; } }

    public class Container<T> {
        private List<Box<T>> items = new List<Box<T>>();
        public void Add(Box<T> item) => this.items.Add(item);
        public int Size() => this.items.Count;
    }

    public static class Helper {
        public static void Log(object o) => Console.WriteLine(o);
    }

    public static class Main {
        public static Container<string> Process(Box<string> input) {
            var c = new Container<string>();
            c.Add(input);
            Helper.Log(c);
            return c;
        }
    }
}

File: docs/test-beds/scala/fixture.scala
// Scala test bed
package testbed

case class Box[T](value: T)

class Container[T] {
  private val items: scala.collection.mutable.ListBuffer[Box[T]] = scala.collection.mutable.ListBuffer()
  def add(item: Box[T]): Unit = items += item
  def size: Int = items.length
}

object Helper {
  def log(o: Any): Unit = println(o)
}

object Main {
  def process(input: Box[String]): Container[String] = {
    val c = new Container[String]()
    c.add(input)
    Helper.log(c)
    c
  }
}
